fission-server/src/models/email_verification.rs

```diff
@@ -0,0 +1,154 @@
+//! Email Verification Model
+use openssl::sha::Sha256;
+use serde::Deserialize;
+use std::collections::HashMap;
+use tracing::log;
+use utoipa::ToSchema;
+use validator::{Validate, ValidationError};
+
+use anyhow::Result;
+
+use mailgun_rs::{EmailAddress, Mailgun, Message};
+
+use rand::Rng;
+
+use crate::settings::Settings;
+
+/// Generate a code that can be sent to the user.
+fn generate_code() -> u32 {
+    let mut rng = rand::thread_rng();
+    // This is maybe way too little entropy. That said, my bank sends me 5 digit codes. 🤷‍♂️
+    rng.gen_range(10000..=99999)
+}
+
+/// [Request] Parameters
+#[derive(Deserialize, Validate, Clone, Debug, ToSchema)]
+pub struct Request {
+    /// The email address of the user signing up
+    #[validate(email)]
+    pub email: String,
+    /// The (pre-generated) did of the client application.
+    /// Currently only did:key is supported.
+    pub did: String,
+    #[serde(skip)]
+    #[serde(default = "generate_code")]
```

Comment:
It's auto-filled on deserializing, and since we skip deserializing for `code`, serde automatically and always fills it in for us, which is handy, though perhaps too magical?